### PR TITLE
fix: added dns content validation

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Added dns_record validation to avoid conflict between`content` and `data`. 

https://github.ibm.com/GoldenEye/issues/issues/5670
https://github.com/terraform-ibm-modules/terraform-ibm-cis/issues/350

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
The data block and content should not be added together for any DNS record so to avoid conflict a validation has been added for all the different types of DNS records.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by maheshwarishikha** [comment]
@iamar7 I understand that `data` and `content` conflicts with each other i.e. both cant be defined for any scenario. Please confirm on this.

If its true, then can we have a condition which checks that both are not defined irrespective of `record_type`? Because if there is a scenario that user defined `content` and `data` both for `A` record type...then our condition wont be able to catch the error. Please recheck. Thanks

**Comment by iamar7** [comment]
> @iamar7 I understand that `data` and `content` conflicts with each other i.e. both cant be defined for any scenario. Please confirm on this.
> 
> If its true, then can we have a condition which checks that both are not defined irrespective of `record_type`? Because if there is a scenario that user defined `content` and `data` both for `A` record type...then our condition wont be able to catch the error. Please recheck. Thanks

Modified the validation condition based on the given suggestion.

File: modules/dns/main.tf [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 locals {
   # tflint-ignore: terraform_unused_declarations
-  validate_inputs = [for record in var.dns_record_set : (record.type == "SRV" || record.type == "LOC" || record.type == "CAA") ? (record.data != null ? null : tobool("The data{} block in dns_record_set is not defined for the DNS record-  ${record.name}")) : null]
+  validate_dns_record_data = [for record in var.dns_record_set : (record.type == "SRV" || record.type == "LOC" || record.type == "CAA") ? (record.data != null ? null : tobool("The data{} block in dns_record_set is not defined for the DNS record- ${record.name}")) : null]
+  # tflint-ignore: terraform_unused_declarations
+  validate_dns_record_content = [for record in var.dns_record_set : ((record.content == null && record.data != null) || (record.content != null && record.data == null)) ? null : tobool("The content argument and data{} block in dns_record_set should not be defined together for the DNS record- ${record.name}")]
```
[the remark this answers is not in the document]
Comment (iamar7) [reply modules/dns/main.tf]: Updated the message.

**Pull request merged**